src/coreclr/src/jit/lsraxarch.cpp

```diff
@@ -2661,6 +2647,11 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
             {
                 srcCount += BuildAddrUses(op1);
             }
+            else if (isRMW && !op1->isContained())
```

Comment:
This and the logic just below was largely taken from `BuildRMWUses`.

I wasn't able to reuse it due to there being 3 operands and various additional handling needed for the HWIntrinsics.